test/e2e_node/device_plugin.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"strings"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// dummy device file saved in "/tmp/dummy-device" directory,
+	// e.g. "/tmp/dummy-device/device-1", "/tmp/dummy-device/device-2"
+	dummyDeviceDir string = "dummy-device"
+	dummyDeviceRE  string = "^device-[0-9]*$"
+)
+
+// Serial because the test restarts Kubelet
+var _ = framework.KubeDescribe("Device Plugin [Feature:DevicePlugin] [Serial] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("device-plugin-errors")
+
+	Context("DevicePlugin", func() {
+		By("Enabling support for Device Plugin")
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.FeatureGates[string(features.DevicePlugins)] = true
+		})
+
+		BeforeEach(func() {
+			By("Create dummy device files on the node")
+			if err := framework.CreateDummyDeviceFileFromConfig(); err != nil {
+				Skip("Faild to create dummy device files. Skipping test.")
+			}
+
+			By("Creating stub device plugin pod")
+			f.PodClient().CreateSync(framework.StubDevicePlugin(f.Namespace.Name))
+
+			// TODO(cph): remove sleep here will break test
+			time.Sleep(30 * time.Second)
```

Comment:
Can we use an `Eventually` here ?

Reply:
or can we just remove this `Sleep` as there is an `Eventually` just after this which is looking for devices availability?

Reply:
@RenaudWasTaken @vikaschoudhary16 Thanks so much for your comments.
Yes, you are right, I had originally planned to remove `Sleep`, but this function `getLocalNode` will return error: `Reason: KubeletNotReady, message: container runtime is down` to break test.

I add TODO and will dig this further.

Reply:
@jiayingz After debugging, i think this is because `tempSetCurrentKubeletConfig` will restart `kubelet`.

I add Eventually `wait for node is ready` here.